charts/backstage/templates/backstage-deployment.yaml

```diff
@@ -25,9 +25,7 @@ spec:
       labels: {{- include "common.labels.standard" . | nindent 8 }}
         app.kubernetes.io/component: backstage
     spec:
-      {{- if .Values.serviceAccount.create }}
-      serviceAccountName: {{ .Values.serviceAccount.name }}
-      {{- end }}
+      serviceAccountName: {{ include "backstage.serviceAccountName" . }}
```

Comment:
so this should potentially fail under the condition of:

```
serviceAccount:
  create: false
  name: 'foobar'
```

as foobar service account doesn't exist it shouldn't be able to startup the container correct? This is something a user will do and while its a easy fix, its also easy to prevent someone doing something silly

Reply:
If a service account is defined, but does not exist, the pod will not be scheduled. The goal of this PR is to enable users to be able to leverage existing service accounts that are defined in the cluster. It is possible to perform validation of existing resources using the `lookup` Helm function. However, several tools (including Argo CD) use the `template` function of helm which does not cause the `lookup` function to be triggered. Kubernetes events are emitted for service account that are declared, but not found making it easy for the user to rectify

Reply:
@sabre1041 describes the goal well. Our typical use case is that we provision service accounts in namespaces that have pre-defined image pull secrets to access a private registry as well as a custom RBAC roles.

The PR is doing the same thing as well known charts, such as the ones from bitnami (e.g. https://github.com/bitnami/charts/blob/91fca1180126322b3d9ddc748ebe48de53f139fe/bitnami/postgresql/values.yaml#L1169).

I don't think I've seen charts validating the presence of the service account in case you use `create: false` (I think the FailedCreated event is pretty clear in this case).

Reply:
I agree that we should rather prefer not using lookups here for sake of versatility of this tool.

Also events generated and failure to deploy is a reasonable fail-safe here. If user intentionally leaves create as `false` yet specifies a service account name, I think it's safe to assume they want this behavior. And if they don't the fail state is reasonably advertised and easy to debug.